### ISSUE TITLE
How do I put a specific order to Jbehave story execution in Serenity BDD?

### ISSUE THREAD

**Issue by Santhosh17** [serenity-bdd/serenity-jbehave#101]
I have created some story files and whenever i try to execute it. it executes in alphabetical order is there a way to control the execution in the order which we specify.
ex: current execution
aaa.story
bbb.story
ccc.story

required execution
ccc.story
aaa.story
and skip bbb.story

**Comment by wakaleo** [comment]
The order of execution is controlled by JBehave. As a general rule, you should try to write your stories in a way so that the order of execution doesn't matter - if one story needs logic to have been executed from another story, for example, place it in a reusable step, rather then depending on the story to have already been run.

**Comment by im-zeeshan** [comment]
As a general rule, story or feature files should contain a collection of Test Scenarios which test a common functionality. Test Scenarios within a story or feature file can be reordered/sequenced to achieve the desired testing goal. 
Story or feature files should test independent modules and should not have any inter-dependency of execution. This not just applies to automation testing but for manual as well. If there were inter-dependency of story or feature files, it will become very difficult to manage the testing since the tests are fragmented across various files.

**Comment by wakaleo** [comment]
Thanks for your comment. It is indeed important to ensure that tests are independent, for the reasons you mention. 

I'm not sure I follow your comment on "test scenarios a story or feature file can be reordered/sequenced to achieve the desired testing goal". In Behaviour Driven Development (and tools like JBehave and Cucumber are built on this assumption), a scenario typically represents a concrete example that illustrates a particular way the user interacts with the system to achieve the goal described by the feature - for example with different inputs and expected outcomes. So if a feature is "pay by credit card", a scenario might  be "pay by Visa", but it would not typically be "enter credit card number", "enter address", or "validate payment details". So scenarios, too, can be executed in any order.

**Comment by andriyyarish** [comment]
The main flow for such dependent scenarios step is to test sequence that perform some CRUD flow. And if Create scenario is a heavy and time-consuming operation.  So specifying the order in such case should be very useful feature.

**Comment by wakaleo** [comment]
If you have heavy initial operations (such as data injection) that are used by all the scenarios, it is probably better to handle this "under the hood", through a lazy loading mechanism for example, than depending on story or scenario ordering. Scenarios in any of the Gherkin-based tools are really not designed to be used in that way.

I'll close this issue as it is a JBehave question rather than something specific to Serenity.
